app/components/server.ts

```diff
@@ -14,4 +18,90 @@ import { JobRequest } from '@/api'
     }
   })
 export const sendRequest = async (jr: JobRequest) => server.queueJob(jr)
-// end TODO
\ No newline at end of file
+// end TODO
+
+
+export const getServerTelegramClient = async (sessionString: string): Promise<TelegramClient> => {
+  return await getTelegramClient(sessionString)
+}
+
+export const getMe = async (sessionString: string): Promise<string> => {
+	const client = await getTelegramClient(sessionString)
+
+	if(!client.connected){
+		await client.connect()
+	}
+
+	const user = await client.getMe()
+	return user.id.toString()
+}
+
+export const listDialogs = async (sessionString: string, paginationParams: { limit: number, offsetId?: number, offsetDate?: number, offsetPeer?: string}) => {
```

Comment:
do we have a sense of the implications of doing this whole fetch on the server before returning any data to the client? it would be useful to see how long this delays display of chats for a user with lots of data -- like @prodalex  for example.

Reply:
I going to work on an improvement with a "load more" logic in another PR.